src/hotspot/share/opto/vectornode.cpp

```diff
@@ -1775,6 +1791,95 @@ Node* ReverseVNode::Identity(PhaseGVN* phase) {
   return this;
 }
 
+Node* AndVNode::Identity(PhaseGVN* phase) {
+  // (AndV src (Replicate m1))   => src
+  // (AndVMask src (MaskAll m1)) => src
+  if (VectorNode::is_all_ones_vector(in(2))) {
+    return in(1);
+  }
+  // (AndV (Replicate zero) src)   => (Replicate zero)
+  // (AndVMask (MaskAll zero) src) => (MaskAll zero)
+  if (VectorNode::is_all_zeros_vector(in(1))) {
```

Comment:
Why you expect it to be `in(1)` instead of `in(2)` as in previous case? Do we create inputs in such order based on mask value?
At least add comment explaining it.

Reply:
We also have the case that the all zeros vector is `in(2)` in the followed codes. Please see line 1817. The main reason to do the different handle is the consideration for the predicated vector operations in Vector API.

Reply:
That is what confuses me. The comment there says: `masked operation requires the unmasked lanes to save the same values in the first operand`.

I'm interpreting it as mask should be `in(2)`:
https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/aarch64/aarch64_sve.ad#L379

But here you check `in(1)`.

Reply:
The comment means: for masked operations, the result of non-masked lanes should be from `in(1)`, and the masked lanes are from the operation results. 
For `"AndV"` with zero, the results is zero. So if `in(1)` is all zeros vector which is the expected result, no matter whether the `AndV` is masked or not, the result is right (i.e. for masked `AndV`, the non-masked lanes should be from `in(1)`, and the masked lanes should be from the operation result which is also `in(1)`).  But if the all zeros vector is `in(2)`, this transformation will the results not right for masked `AndV`. That's why I added `!is_predicated_vector()` limit for the second case while not for the first one.

Reply:
> The comment means: for masked operations, the result of non-masked lanes should be from `in(1)`, and the masked lanes are from the operation results.

Okay, I got it finally. Thank you for explanation.